runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5228,6 +5228,7 @@ class INTERPRETER_CLASS
 					restoreInternalNativeStackFrame(REGISTER_ARGS);
 					/* Handle the virtual thread Object.wait call. */
 					J9VMJAVALANGVIRTUALTHREAD_SET_NOTIFIED(_currentThread, _currentThread->threadObject, JNI_FALSE);
+					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, millis);
```

Comment:
Can you provide the context for the `VirtualThread.timeout` field? Outside the native code, how is it managed on the Java side? Do we need to worry about resetting the timeout field?

Reply:
`VirtualThread.timeout` is a private field used by both VM and JCL to temporarily hold the value of expected wait/park time before a wake up task is scheduled using the value.
there is no need to reset the field as it should be set before use in all use case.